tests/replication/_base.py

```diff
@@ -69,7 +69,6 @@ def prepare(self, reactor: MemoryReactor, clock: Clock, hs: HomeServer) -> None:
         # Make a new HomeServer object for the worker
         self.reactor.lookups["testserv"] = "1.2.3.4"
         self.worker_hs = self.setup_test_homeserver(
-            federation_http_client=None,
```

Comment:
This change causes a failure with `tests.replication.tcp.streams.test_federation.FederationStreamTestCase.test_catchup`


https://github.com/matrix-org/synapse/actions/runs/5271355310/jobs/9532147858?pr=15773#step:9:4325
```
Error: 
Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
DelayedCalls: (set twisted.internet.base.DelayedCall.debug = True to debug)
<DelayedCall 0x7f7bfa3c3a50 [59.65530204772949s] called=0 cancelled=0 Resolver.maybeParseConfig()>

tests.replication.tcp.streams.test_federation.FederationStreamTestCase.test_catchup
```

```
SYNAPSE_POSTGRES=1 SYNAPSE_POSTGRES_USER=postgres SYNAPSE_TEST_LOG_LEVEL=INFO python -m twisted.trial tests.replication.tcp.streams.test_federation.FederationStreamTestCase.test_catchup
```

Reply:
I think(?) this means that there was some deferred that wasn't awaited at the end of the test.